cloud/ec2_util.go

```diff
@@ -472,3 +473,68 @@ func IsEc2Provider(provider string) bool {
 		provider == evergreen.ProviderNameEc2Spot ||
 		provider == evergreen.ProviderNameEc2Fleet
 }
+
+// Get EC2 region from an EC2 ProviderSettings object
+func GetEC2Region(providerSettings *map[string]interface{}) string {
+	s := &EC2ProviderSettings{}
+	if providerSettings != nil {
+		if err := mapstructure.Decode(providerSettings, s); err != nil {
+			return evergreen.DefaultEC2Region
+		} else {
+			return s.getRegion()
+		}
+	}
+
+	return evergreen.DefaultEC2Region
+}
+
+// Get EC2 key and secret from the AWS configuration for the given region
+func GetEC2Key(region string, s *evergreen.Settings) (string, string, error) {
+	// Get default region if field is blank
+	if region == "" {
+		region = evergreen.DefaultEC2Region
+	}
+
+	// Get key and secret for specified region
+	var key, secret string
+	found := false
```

Comment:
Rather than using found, I think you should check if key/secret are empty in both places at the time you find them and return early if valid and error if not valid. The tiny piece of repeated code isn't a problem since legacy will be deleted soon anyway.